From f3bdcf9a9d835dc20abf7a49333a23261ee2373e Mon Sep 17 00:00:00 2001
From: Jerome Duval <jerome.duval@gmail.com>
Date: Mon, 30 Jan 2017 20:01:22 +0100
Subject: x86_64 build fix.

* have build.sh exit on failure.
* use build.sh parameters as parameters for make

diff --git a/addons/AddOns/Working/WaveAddOn/Wave.h b/addons/AddOns/Working/WaveAddOn/Wave.h
index ce95577..50858b3 100644
--- a/addons/AddOns/Working/WaveAddOn/Wave.h
+++ b/addons/AddOns/Working/WaveAddOn/Wave.h
@@ -135,7 +135,7 @@ inline 	asm	float reciprocal_of_square_root(register float number)
 	frsqrte		fp1,number;	// Estimates reciprocal of square-root
 	blr
 }
-#elif __INTEL__
+#elif __INTEL__ || defined(__x86_64__)
 float reciprocal_of_square_root(register float number)
 {
 	return 1.0 / sqrt(number);
diff --git a/artpaint/controls/ColorPalette.cpp b/artpaint/controls/ColorPalette.cpp
index 2b103a7..dd199f7 100644
--- a/artpaint/controls/ColorPalette.cpp
+++ b/artpaint/controls/ColorPalette.cpp
@@ -1134,7 +1134,7 @@ void ColorContainer::MessageReceived(BMessage *message)
 			// Here we see on to which button it was dropped and then
 			// try to extract a color from the message
 			rgb_color *color;
-			int32 color_size;
+			ssize_t color_size;
 			if (message->FindData("RGBColor",B_RGB_COLOR_TYPE,(const void**)&color,&color_size) == B_OK) {
 				BPoint drop_point = message->DropPoint();
 				drop_point = ConvertFromScreen(drop_point);
diff --git a/artpaint/controls/ColorView.cpp b/artpaint/controls/ColorView.cpp
index acc4105..76e4c59 100644
--- a/artpaint/controls/ColorView.cpp
+++ b/artpaint/controls/ColorView.cpp
@@ -64,7 +64,7 @@ void ColorView::Draw(BRect)
 void ColorView::MessageReceived(BMessage *message)
 {
 	rgb_color *new_color;
-	int32 color_size = sizeof(rgb_color);
+	ssize_t color_size = sizeof(rgb_color);
 	switch (message->what) {
 		case B_PASTE:
 			if (message->FindData("RGBColor",B_RGB_COLOR_TYPE,(const void**)&new_color,&color_size) == B_OK) {
diff --git a/artpaint/controls/VisualColorControl.cpp b/artpaint/controls/VisualColorControl.cpp
index f053728..dab2b7b 100644
--- a/artpaint/controls/VisualColorControl.cpp
+++ b/artpaint/controls/VisualColorControl.cpp
@@ -193,7 +193,7 @@ void VisualColorControl::Draw(BRect)
 void VisualColorControl::MessageReceived(BMessage *message)
 {
 	rgb_color *new_color;
-	int32 color_size;
+	ssize_t color_size;
 	switch (message->what) {
 		case B_PASTE:
 			if (message->FindData("RGBColor",B_RGB_COLOR_TYPE,(const void**)&new_color,&color_size) == B_OK) {
diff --git a/artpaint/paintwindow/StatusView.cpp b/artpaint/paintwindow/StatusView.cpp
index 4f306f2..533979d 100644
--- a/artpaint/paintwindow/StatusView.cpp
+++ b/artpaint/paintwindow/StatusView.cpp
@@ -452,7 +452,7 @@ SelectedColorsView::MessageReceived(BMessage *message)
 				// Here we see on to which button it was dropped and then
 				// try to extract a color from the message
 				rgb_color *color;
-				int32 color_size;
+				ssize_t color_size;
 				if (message->FindData("RGBColor",B_RGB_COLOR_TYPE,
 						(const void**)&color, &color_size) == B_OK) {
 					BPoint drop_point = message->DropPoint();
diff --git a/artpaint/windows/BrushStoreWindow.cpp b/artpaint/windows/BrushStoreWindow.cpp
index 89ad4be..11c0b71 100644
--- a/artpaint/windows/BrushStoreWindow.cpp
+++ b/artpaint/windows/BrushStoreWindow.cpp
@@ -333,7 +333,7 @@ void BrushStoreView::MessageReceived(BMessage *message)
 {
 	brush_info *info;
 	Brush *a_brush;
-	int32 size;
+	ssize_t size;
 	switch (message->what) {
 		case HS_BRUSH_DRAGGED:
 			if ((message->ReturnAddress() == BMessenger(this)) == FALSE) {
diff --git a/build.sh b/build.sh
index 71cd0d3..957c998 100755
--- a/build.sh
+++ b/build.sh
@@ -1,9 +1,10 @@
+#!/bin/sh -e
 echo "Building ArtPaint"
-make -f Makefile OBJ_DIR="objects_artpaint"
+make -f Makefile OBJ_DIR="objects_artpaint" $*
 echo "Building ArtPaint addons"
-make -f Makefile_addons OBJ_DIR="objects_addons"
+make -f Makefile_addons OBJ_DIR="objects_addons" $*
 echo "Moving final executable and addons into dist folder"
 mkdir -p dist/add-ons
 cp -af objects_artpaint/ArtPaint dist
 cp -rf addons/AddOns/Working/*/objects_addons/*.so dist/add-ons
-echo "Build has finished!"
\ No newline at end of file
+echo "Build has finished!"
-- 
2.10.2

